### PR TITLE
Refactor/sgc prompt

### PR DESCRIPTION
**Pull request by rudolfsonjunior**
* add new file lib/promptConfig.js
* delete file lib/sgcPrompt.js
* add tests for lib/promptConfig
* delete tests for lib/sgcPrompt

### CODE REVIEW

File: test/prompConfig.js [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+const cwd = process.cwd();
+
+test('get configuration file equals .sgcrc_default', (t) => {
+  const cli = promptConfig;
+  const config = cli.config();
+  t.deepEqual(config, json.readToObjSync(path.join(cwd, '.sgcrc_default')));
+});
+
+test('choises are the same as choises generated from .sgcrc_default', (t) => {
```
Comment (dani-sc) [inline test/prompConfig.js]: choises -> choices

File: test/prompConfig.js [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+
+test('check the values of the question object', (t) => {
+  const cli = promptConfig;
+  const config = cli.config();
+  const choices = cli.choices(config);
+  const questions = cli.questions(choices);
+
+  t.deepEqual(questions[0].type, 'list');
+  t.deepEqual(questions[0].name, 'type');
+  t.deepEqual(questions[0].message, 'Select the type of your commit:');
```
Comment (dani-sc) [inline test/prompConfig.js]: Should we really unit test the _message_s? In this case, if we change something, we immediately break a unit test, even though it's actually still valid.

File: lib/promptConfig.js [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+        return execa('git', ['commit', '-m', message])
+          .then(result => console.log(result.stdout))
+          .catch((err) => {
+            console.error(chalk.red("Have you thought about 'git add' some files? Add files and try run following again:\n"));
+            console.error(chalk.green(err.cmd));
+          });
+      }
+      return answers.type;
+    });
+    return cli;
```
Comment (JPeer264) [inline lib/promptConfig.js]: maybe `cli` is in that context a bad name. Maybe `prompt` fits better, as it is actually a prompt

File: lib/promptConfig.js [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+  },
+
+  answerTypes: (configuration) => {
+    const configTypeList = [];
+    configuration.types.forEach((type) => {
+      const configType = type.type;
+
+      configTypeList.push(configType);
+    });
+    return configTypeList;
```
Comment (JPeer264) [inline lib/promptConfig.js]: as in `choices` there should be a break/whitespace before the return

File: lib/promptConfig.js [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+      },
+      {
+        type: 'editor',
+        name: 'editor',
+        message: 'This will let you add more information',
+        when: /* istanbul ignore next */ answers => answers.moreInfo,
+        default: /* istanbul ignore next */ answers => `${answers.type} ${answers.description}\n\n\n`,
+      },
+    ];
+    return questionsList;
```
Comment (JPeer264) [inline lib/promptConfig.js]: break before return

File: lib/promptConfig.js (hunk 3 above)
Comment (JPeer264) [inline lib/promptConfig.js]: break before return

File: lib/promptConfig.js [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+        type: 'list',
+        name: 'type',
+        message: 'Select the type of your commit:',
+        choices,
+      },
+      {
+        type: 'input',
+        name: 'description',
+        message: 'Enter your commit message:',
+        validate: /* istanbul ignore next */ input => (input ? true : 'A commit message is mandatory!'),
```
Comment (JPeer264) [inline lib/promptConfig.js]: Just curious, why does the next line get ignored?

File: test/prompConfig.js (hunk 2 above)
Comment (rudolfsonjunior) [inline test/prompConfig.js]: Yeah thats a good point. Only the messages or do think the whole questions?

File: test/prompConfig.js (hunk 1 above)
Comment (rudolfsonjunior) [inline test/prompConfig.js]: I will fix that.

File: lib/cli.js [hunk 7]
```diff
@@ -1,5 +1,9 @@
 #!/usr/bin/env node
-// import { argv } from 'yargs';
-import sgcPrompt from './sgcPrompt';
+import promptConfig from './promptConfig';
```
Comment (JPeer264) [inline lib/cli.js]: How about using models? It is recommended by the [airbnb styleguide](https://github.com/airbnb/javascript#modules--use-them).

`import { config, ... } from './promptConfig'`

File: lib/promptConfig.js [hunk 8]
```diff
@@ -0,0 +1,87 @@
...
+        name: `${chalk.bold(configType)} ${description}`,
+      });
+    });
+
+    return choicesList;
+  },
+
+  answerTypes: (configuration) => {
+    const configTypeList = [];
+    configuration.types.forEach((type) => {
```
Comment (JPeer264) [inline lib/promptConfig.js]: Breaks are also quite important after `const` `let`

File: lib/promptConfig.js [hunk 9]
```diff
@@ -0,0 +1,87 @@
...
+    return questionsList;
+  },
+
+  prompt: (questions) => {
+    const config = promptConfig.config();
+    const types = promptConfig.answerTypes(config);
+    const cli = inquirer.prompt(questions).then((answers) => {
+      const message = answers.moreInfo ? `${answers.editor}` : `${answers.type} ${answers.description}`;
+      /* istanbul ignore if */
+      if (types.indexOf(answers.type.split(' ')[1]) !== -1) {
```
Comment (JPeer264) [inline lib/promptConfig.js]: I don't really get the if. Can you explain it?

File: test/prompConfig.js (hunk 2 above)
Comment (dani-sc) [inline test/prompConfig.js]: Yeah, you're right, we could remove the whole questions with the same reasoning 👍

Comment (JPeer264) [inline test/prompConfig.js]: I think non of the questions should be tested. The only thing what you can test is:

- Is the return of `questions` an array
- is there at least one object inside

File: lib/promptConfig.js (hunk 9 above)
Comment (rudolfsonjunior) [inline lib/promptConfig.js]: Only if a valid type exists you can commit your message. If the user is writing a invalid .`sgcrc`file. Otherwise it should return a error. Maybe we should rewrite it with `if (answers.type){ ... } else { ... }`

Comment (JPeer264) [inline lib/promptConfig.js]: When is a type invalid? 

For the validation:
Maybe we can validate the config before in `loadConfig`. So we can fail fast and we don't have to abort after writing all messages.

File: lib/promptConfig.js (hunk 6 above)
Comment (rudolfsonjunior) [inline lib/promptConfig.js]: I tried something. I will remove it.

File: lib/promptConfig.js (hunk 3 above)
Comment (rudolfsonjunior) [inline lib/promptConfig.js]: Ok

File: lib/promptConfig.js (hunk 9 above)
Comment (rudolfsonjunior) [inline lib/promptConfig.js]: if you have a 
`type`
in your sgcrc file. the way you specified it ;) the prefix before refactoring. Then a type is valid.

Then we should definitely not test the prompt function. Because it will make a commit every time.

Comment (JPeer264) [inline lib/promptConfig.js]: Alright, we should make a validation in `getConfig`. Maybe as `validateConfig` and rename the file `getConfig.js` to `config.js` or similar? What you think?

Maybe we can test it, I will do a research about that.

Comment (rudolfsonjunior) [inline lib/promptConfig.js]: Yes. That is a good idea. 

But prompt should land into cli.js, because we should not test the functionality of a library we are using.

File: lib/cli.js (hunk 7 above)
Comment (rudolfsonjunior) [inline lib/cli.js]: If it is recommended I will definitely change it.

File: lib/promptConfig.js (hunk 9 above)
Comment (JPeer264) [inline lib/promptConfig.js]: Alright, the validation could be in another PR. So we can remove the `if` from this PR?

Good idea. Lets move prompt into `cli.js`

File: lib/cli.js (hunk 7 above)
Comment (JPeer264) [inline lib/cli.js]: 👍

File: test/prompConfig.js (hunk 2 above)
Comment (rudolfsonjunior) [inline test/prompConfig.js]: I will change that test.

File: lib/promptConfig.js (hunk 9 above)
Comment (rudolfsonjunior) [inline lib/promptConfig.js]: PR?

Comment (JPeer264) [inline lib/promptConfig.js]: pull request :)

Comment (rudolfsonjunior) [inline lib/promptConfig.js]: Thanks ^^ okay then i just move the prompt into cli and remove the if stmt.

Comment (JPeer264) [inline lib/promptConfig.js]: 👍

File: lib/promptConfig.js [hunk 10]
```diff
@@ -1,87 +1,55 @@
 import chalk from 'chalk';
-import execa from 'execa';
-import inquirer from 'inquirer';
 import getConfig from './getConfig';
 
-const promptConfig = {
-  config: () => getConfig(),
+const config = () => getConfig();
```
Comment (JPeer264) [inline lib/promptConfig.js]: I just saw it. I guess we can remove that and call it directly from `getConfig()`

Comment (rudolfsonjunior) [inline lib/promptConfig.js]: got it

**Comment by JPeer264** [comment]
Thanks @rudolfsonjunior

**Comment by rudolfsonjunior** [comment]
Had to fix a bug!
